### PR TITLE
gh-117474: Skip GIL test in free-threaded build

### PR DESCRIPTION
**Pull request by colesbury**
In the free-threaded build, the GIL will typically be disabled so `py-bt` will not show threads waiting on the GIL.

* Issue: gh-117474

### CODE REVIEW

**Comment by colesbury** [comment]
!buildbot nogil

File: Lib/test/test_gdb/test_backtrace.py [hunk 1]
```diff
@@ -49,6 +49,7 @@ def test_bt_full(self):
 
     @unittest.skipIf(python_is_optimized(),
                      "Python was compiled with optimizations")
+    @unittest.skipIf(support.Py_GIL_DISABLED, "test requires the GIL")
```
Comment (Eclips4) [inline Lib/test/test_gdb/test_backtrace.py]: Maybe it's time to introduce a helper in the `support` package like `requires_gil_enabled`? Actually I don't know how many cases there are  in tests like this, if it's just one case, there's no need for a helper.

Comment (colesbury) [reply Lib/test/test_gdb/test_backtrace.py]: Including this one, there are six `@unittest.skipIf(support.Py_GIL_DISABLED, ...)`

A helper seems like a reasonable idea to me.

Comment (Eclips4) [reply Lib/test/test_gdb/test_backtrace.py]: > Including this one, there are six `@unittest.skipIf(support.Py_GIL_DISABLED, ...)`
> 
> A helper seems like a reasonable idea to me.

I think this helper should look like this:
```python
def requires_gil_enabled(test):
    if Py_GIL_DISABLED:
        return unittest.skip('needs the GIL enabled')(test)
    return test
```

Comment (colesbury) [reply Lib/test/test_gdb/test_backtrace.py]: Thanks! I'll add this in a separate PR.

**Review by DinoV — APPROVED** [review]
LGTM!

**Pull request merged**